src/main/java/com/endava/internship/collections/StudentMap.java

```diff
@@ -1,72 +1,405 @@
+                    lastNode.setNext(null);
+                }
+                putNode(oldTable[i]);
+                oldTable[i] = null;
+            }
+        }
+    }
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return size == 0;
     }
 
     @Override
     public boolean containsKey(Object o) {
-        //TODO
+        if(o instanceof Student || o == null) {
+            Node node = getNode(new Node(hash(((Student) o)), ((Student) o), null, null));
+            return node != null;
+        }
         return false;
     }
 
     @Override
     public boolean containsValue(Object o) {
-        //TODO
+        if(table != null && size > 0 &&
+                (o instanceof Integer || o == null)) {
+            for(int i = 0; i < table.length; i++) {
+                for(Node node = table[i]; node != null; node = node.getNext()) {
+                    if(node.getVal() == o || node.getVal().equals(o)) {
+                        return true;
+                    }
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public Integer get(Object o) {
-        //TODO
+        if(o instanceof Student || o == null) {
+            Node node = getNode(new Node(hash(((Student) o)), ((Student) o), null, null));
+            return node == null ? null : node.getVal();
+        }
+        return null;
+    }
+    private Node getNode(Node node) {
+        if(table != null && size > 0) {
+            Node listNode = node.getKey() == null ? table[0] :
+                    table[(table.length - 1) & node.getHash()];
+            if(listNode != null) {
+                while(true) {
+                    if(listNode.equals(node)) {
+                        node.setVal(listNode.getVal());
+                        return node;
+                    }
+                    else {
+                        if(listNode.getNext() == null) {
+                            break;
+                        }
+                        else {
+                            listNode = listNode.getNext();
+                        }
+                    }
+                }
+            }
+        }
         return null;
     }
-
     @Override
     public Integer put(Student student, Integer integer) {
-        //TODO
-        return null;
+        return putNode(new Node(hash(student), student, integer, null));
     }
 
+    private Integer putNode(Node node) {
+        if(table == null) {
+            table = new Node[capacity];
+        }
+        if(size > table.length * loadFactor) {
+            resize();
+        }
+        int i = node.getKey() == null ? 0 : (table.length - 1) & node.getHash();
+        if(table[i] == null) {
+            table[i] = node;
+            size++;
+        }
+        else {
+            // Идти по ссылкам
```

Comment:
usually we don't leave any comments, if there is no important reason or explanation why it should be implimented this way